pyqtorch/modules/parametric.py

```diff
@@ -81,9 +81,9 @@ def __init__(self, qubits: ArrayLike, n_qubits: int):
 
     def matrices(self, thetas: torch.Tensor) -> torch.Tensor:
         if thetas.ndim == 1:
-            thetas = thetas.unsqueeze(1)
+            thetas = thetas.unsqueeze(1).cdouble()
         assert thetas.size(0) == 3
-        phi, theta, omega = thetas[0, :], thetas[1, :], thetas[2, :]
+        phi, theta, omega = thetas[0, :].cdouble(), thetas[1, :].cdouble(), thetas[2, :].cdouble()
```

Comment:
hmm, a bit ugly to always have to call this, isnt it? can we maybe find a better way. maybe there is something in the torch docu which enables the required precision. to me it seems like that it should maybe be handled on the user-side which passes the thetas?

Reply:
I agree it's ugly yes. We could try to do `torch.set_default_dtype(torch.float64) like documented here: https://pytorch.org/docs/stable/generated/torch.set_default_dtype.html.

I guess in theory it might allows us to remove all the `cdouble` both from the thetas and maybe also from the `dtype = cdouble` when initializing torch tensors, but I assume this would have to be added to every single file in the library.

For the theta's here the reason I don't think it should be handled by the user is that for any of these parametric gates if the user just sets "angle = 1.0" or something with python float, which seems very natural, it is currently cast to 32 bit precision. Even just one gate that does this will ruin the precision of the whole circuit.

Reply:
Alright I've tested it now and `torch.set_default_dtype(torch.float64)` seems to do the trick! I removed the `.cdouble()` I added (but not the ones from the torch tensor initializations).

Would it make sense to have this line added by default when linting, or you think that could be dangerous in some way?

Reply:
yeah we should test it on GPU for sure